homeassistant/components/blebox/cover.py

```diff
@@ -0,0 +1,129 @@
+"""BleBox cover entity."""
+
+import logging
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    DEVICE_CLASS_DOOR,
+    DEVICE_CLASS_SHUTTER,
+    STATE_CLOSED,
+    STATE_CLOSING,
+    STATE_OPEN,
+    STATE_OPENING,
+    SUPPORT_CLOSE,
+    SUPPORT_OPEN,
+    SUPPORT_SET_POSITION,
+    SUPPORT_STOP,
+    CoverDevice,
+)
+from homeassistant.exceptions import PlatformNotReady
+
+from . import BleBoxEntity, async_add_blebox
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add):
+    """Set up a BleBox entry."""
+    return await async_add_blebox(
+        BleBoxCoverEntity,
+        "covers",
+        hass,
+        config_entry.data,
+        async_add,
+        PlatformNotReady,
```

Comment:
```suggestion
```
Please raise `ConfigEntryNotReady` in `__init__` if there is a connection failure instead

Reply:
As above, ConfigEntryNotReady doesn't seem to work in practice.

(I'll refactor this out once the ConfigEntryNotReady/PlatformNotReady issue is figured out).